plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -131,13 +135,42 @@ def save(self):
             except IntegrityError:
                 is_duplicate = True
             else:
-                for artifact in self.artifacts:
-                    artifact.model.content = self.model
-                    artifact.save()
-        if is_duplicate:
-            model = type(self.model)
-            content = model.objects.get(**self.key)
-            self.update(content)
+
+                for remote_artifact in self.artifacts:
+                    if not remote_artifact.content_artifact.artifact_id:
+                        if not deferred:
+                            artifact = self.validate_and_create_artifact(remote_artifact.model,
+                                                                         remote_artifact.download)
+                            remote_artifact.content_artifact.artifact = artifact
+                        remote_artifact.model.save()
+                        remote_artifact.content_artifact.save()
+
+    def validate_and_create_artifact(self, remote_artifact, download):
+        """
```

Comment:
I don't know what decisions were made with the epic _validation_ discussion on pulp-dev but seem wrong for platform _core_ components to be using the serializer to do validation and creation.  I don't think we do this anywhere else.